src/error.rs

```diff
@@ -9,29 +8,27 @@ pub enum Error {
 
     /// Error parsing listen address
     AddrParse(net::AddrParseError),
+
+    /// No bytes were read from the stream
+    NoBytesRead,
 }
 
-impl ::std::error::Error for Error {
-    fn cause(&self) -> Option<&dyn std::error::Error> {
+impl error::Error for Error {
```

Comment:
should have called this out in the description as well, but this was using a deprecated method of implementing std::error::Error.

Reply:
heh, i was going to ask. Apparently it's been a while!